src/core/Akka/Actor/ActorPath.cs

```diff
@@ -327,74 +327,165 @@ public static bool TryParse(string path, out ActorPath actorPath)
+        /// </summary>
+        /// <param name="path">The string representation of the <see cref="ActorPath"/>.</param>
+        /// <param name="address">If <c>true</c>, the parsed <see cref="Address"/>. Otherwise <c>null</c>.</param>
+        /// <param name="absoluteUri">A <see cref="ReadOnlySpan{T}"/> containing the path following the address.</param>
+        /// <returns><c>true</c> if the <see cref="Address"/> could be parsed, <c>false</c> otherwise.</returns>
+        private static bool TryParseAddress(string path, out Address address, out ReadOnlySpan<char> absoluteUri)
         {
-            //This code corresponds to AddressFromURIString.unapply
             address = null;
-            if (!Uri.TryCreate(path, UriKind.Absolute, out uri))
+
+            var spanified = path.AsSpan();
+            absoluteUri = spanified;
+
+            var firstColonPos = spanified.IndexOf(':');
+
+            if (firstColonPos == -1) // not an absolute Uri
                 return false;
-            var protocol = uri.Scheme; //Typically "akka"
-            if (!protocol.StartsWith("akka", StringComparison.OrdinalIgnoreCase))
-            {
-                // Protocol must start with 'akka.*
+
+            var fullScheme = SpanHacks.ToLowerInvariant(spanified.Slice(0, firstColonPos));
+            if (!fullScheme.StartsWith("akka"))
+                return false;
+
+            spanified = spanified.Slice(firstColonPos + 1);
+            if (!(spanified[0] == '/' && spanified[1] == '/'))
                 return false;
+
+            spanified = spanified.Slice(2); // move past the double //
+            var firstAtPos = spanified.IndexOf('@');
+            var sysName = string.Empty;
+
+            if (firstAtPos == -1)
+            { // dealing with an absolute local Uri
+                var nextSlash = spanified.IndexOf('/');
+
+                if (nextSlash == -1)
+                {
+                    sysName = spanified.ToString();
+                    absoluteUri = "/".AsSpan(); // RELY ON THE JIT
+                }
+                else
+                {
+                    sysName = spanified.Slice(0, nextSlash).ToString();
+                    absoluteUri = spanified.Slice(nextSlash);
+                }
+
+                address = new Address(fullScheme, sysName);
+                return true;
             }
 
+            // dealing with a remote Uri
+            sysName = spanified.Slice(0, firstAtPos).ToString();
+            spanified = spanified.Slice(firstAtPos + 1);
+
+            /*
+             * Need to check for:
+             * - IPV4 / hostnames
+             * - IPV6 (must be surrounded by '[]') according to spec.
+             */
+            var host = string.Empty;
+
+            // check for IPV6 first
+            var openBracket = spanified.IndexOf('[');
+            var closeBracket = spanified.IndexOf(']');
+            if (openBracket > -1 && closeBracket > openBracket)
```

Comment:
I believe the `[` and `]` characters are illegal for actor names, so that shouldn't be a problem.

https://github.com/akkadotnet/akka.net/blob/c663bc230946644b9d91999d9dbbccf8ebc699f5/src/core/Akka/Actor/ActorPath.cs#L109-L131

Reply:
> One thought is we could pre-check for the slash in the path, and then math based off that index... Would then have to get the slash again later or track it's position as we trim, but would avoid this case if we need to worry about it.

I think this is a good idea - won't hurt performance much, because if the check fails we stop the parse and return `false` anyway. If we are successful we have to retrieve this value anyway in the future.